File: day01/main.rb
def part_one(lines)
  increase, previous = 0, 0

  lines.each_with_index do |line, i|
    increase += 1 if line > previous && i != 0

    previous = line
  end

  puts "part 1: #{increase}"
end

def part_two(lines)
  increase, previous = 0, 0

  lines.each_with_index do |line, i|
    next if i+2 > lines.size - 1

    batch = lines[i..i+2].sum

    increase +=1 if batch > previous && i != 0

    previous = batch
  end

  puts "part 2: #{increase}"
end

def main
  lines = File.readlines("input.txt").map(&:to_i)

  part_one(lines)
  part_two(lines)
end

main

File: day01/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	var singlePrevious int
	var singleIncrease int

	var batchPrevious int
	var batchIncrease int

	var lines []int

	file, _ := os.Open("input.txt")

	defer file.Close()

	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		num, _ := strconv.Atoi(scanner.Text())
		lines = append(lines, num)
	}

	for i := 0; i < len(lines); i++ {
		batch := 0

		for _, j := range lines[i : i+3] {
			batch += j
		}

		if lines[i] > singlePrevious && singlePrevious != 0 {
			singleIncrease++
		}

		if batch > batchPrevious && batchPrevious != 0 {
			batchIncrease++
		}

		singlePrevious = lines[i]
		batchPrevious = batch
	}

	fmt.Printf("part 1: %d\n", singleIncrease)
	fmt.Printf("part 2: %d\n", batchIncrease)
}
